src/NUnitFramework/framework/Internal/Results/TestResult.cs

```diff
@@ -498,24 +498,15 @@ public void RecordException(Exception ex)
             if (ex is NUnitException || ex is TargetInvocationException)
                 ex = ex.InnerException;
 
-            if (ex is AssertionException)
-            {
-                AssertionResults.Add(new AssertionResult(AssertionStatus.Failed, ex.Message, StackFilter.DefaultFilter.Filter(ex.StackTrace)));
-            }
-
             if (ex is ResultStateException)
-            {
-                string message = ex is MultipleAssertException
-                    ? CreateLegacyFailureMessage()
-                    : ex.Message;
-
-                string stackTrace = StackFilter.DefaultFilter.Filter(ex.StackTrace);
-
-                SetResult(((ResultStateException)ex).ResultState, message, stackTrace);
-            }
+                SetResult(
+                    ((ResultStateException)ex).ResultState,
+                    ex.Message, 
+                    StackFilter.DefaultFilter.Filter(ex.StackTrace));
```

Comment:
I'd suggest keeping the scope (`{}`) for readability (since the if block still has more than one line).

Reply:
> Personally, I'm not a fan of that. I think the indentation tells the story well enough. Some people fear errors due to indentation that doesn't actually match the nesting, but I haven't run into problems with it myself.

This is absolutely fine and more of a standardization thing honestly. I know StyleCop has a rule for this but as long as you are not following it in this repo it should be OK to ignore it (after all, the `else` block below suffers from the same issue).